src/data_race.rs

```diff
@@ -542,6 +542,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
         Ok(old)
     }
 
+    /// Perform an conditional atomic exchange with a memory place and a new
+    /// scalar value, the old value is returned.
+    fn atomic_conditional_exchange_scalar(
```

Comment:
Also rename this, conditional_exchange is likely to be confused with compare_exchange. Probably to atomic_min_max_scalar or similar. Unless i am mistaken atomic max and min will always "store" for interactions with happens-before relations and the data race detector so this gives the wrong impression about the functions purpose.

Reply:
Will do, I was not hooked on the name.